mobile/lib/pages/editing/edit.page.dart

```diff
@@ -85,44 +74,42 @@ class EditImagePage extends ConsumerWidget {
               Navigator.of(context).popUntil((route) => route.isFirst),
         ),
         actions: <Widget>[
-          if (image != null)
-            TextButton(
-              onPressed: () async {
-                try {
-                  final Uint8List imageData = await _imageToUint8List(image!);
-                  ImmichToast.show(
-                    durationInSecond: 3,
-                    context: context,
-                    msg: 'Image Saved!',
-                    gravity: ToastGravity.CENTER,
-                  );
-
-                  await PhotoManager.editor.saveImage(
-                    imageData,
-                    title: '${asset!.fileName}_edited.jpg',
-                  );
-                  await ref.read(albumProvider.notifier).getDeviceAlbums();
-                  Navigator.of(context).popUntil((route) => route.isFirst);
-                } catch (e) {
-                  ImmichToast.show(
-                    durationInSecond: 6,
-                    context: context,
-                    msg: 'Error: ${e.toString()}',
-                    gravity: ToastGravity.BOTTOM,
-                  );
-                }
-              },
-              child: Text(
-                'Save to gallery',
-                style: Theme.of(context).textTheme.displayMedium,
+          TextButton(
+            onPressed: isEdited
+                ? () async {
+                    try {
+                      final Uint8List imageData =
+                          await _imageToUint8List(image);
+                      await PhotoManager.editor.saveImage(
+                        imageData,
+                        title: "${asset.fileName.split('.').first}_edited.jpg",
```

Comment:
A filename like `this.is.my.image.jpg` will become `this_edited.jpg` right? Maybe the `path` package can be used here

Reply:
Hi, so I just realized after someone told me that the file for a since my past commit was being incorrectly saved. The file was saving as null_edited. I used the asset's file name but somehow the asset was null all the time, so to fix that small issue I had to completely change the code and structure of the edit page.

Reply:
A filename with dots is still saved incorrectly. I tested it, and `this.is.my.image.jpg` gets renamed to `this_edited.jpg`. You could do this to fix that:

```dart
import 'package:path/path.dart' as p;

title: "${p.withoutExtension(asset.fileName)}_edited.jpg"
```

Reply:
Dang that is very clever of you, I never though I would see a name with dots so I casually did that lol. Thanks! 😅